org-roam-graph.el

```diff
@@ -144,120 +141,109 @@ into a digraph."
+                    :select :distinct [file from]
+                    :from links :inner :join refs :on (and (= links:to refs:ref)
+                                                           (= links:type "cite"))
+                    :where (and (in file selected) (in from selected))])
+           (edges       (org-roam-db-query edges-query))
            (edges-cites (org-roam-db-query edges-cites-query)))
       (insert "digraph \"org-roam\" {\n")
-
       (dolist (option org-roam-graph-extra-config)
-        (insert (concat "  "
-                        (car option)
-                        "="
-                        (cdr option)
-                        ";\n")))
-      (insert (format "  node [%s];\n"
-                      (->> org-roam-graph-node-extra-config
-                           (mapcar (lambda (n)
-                                     (concat (car n) "=" (cdr n))))
-                           (s-join ","))))
-      (insert (format "  edge [%s];\n"
-                      (->> org-roam-graph-edge-extra-config
-                           (mapcar (lambda (n)
-                                     (concat (car n) "=" (cdr n))))
-                           (s-join ","))))
-
+        (insert (org-roam-graph--dot-option option) ";\n"))
+      (dolist (node-type '("node" "edge"))
+        (insert (format " %s [%s];\n" node-type
+                        (mapconcat #'org-roam-graph--dot-option
+                                   (symbol-value
+                                    (intern (concat "org-roam-graph-" node-type "-extra-config")))
+                                   ","))))
       (dolist (node nodes)
         (let* ((file (xml-escape-string (car node)))
                (title (or (caadr node)
                           (org-roam--path-to-slug file)))
                (shortened-title (s-truncate org-roam-graph-max-title-length title))
-               (node-properties (list (cons "label" (s-replace "\"" "\\\"" shortened-title))
-                                      (cons "URL" (concat "org-protocol://roam-file?file="
-                                                          (url-hexify-string file)))
-                                      (cons "tooltip" (xml-escape-string title)))))
+               (node-properties
+                `(("label"   . ,(s-replace "\"" "\\\"" shortened-title))
+                  ("URL"     . ,(concat "org-protocol://roam-file?file=" (url-hexify-string file)))
+                  ("tooltip" . ,(xml-escape-string title)))))
           (insert
-           (format "  \"%s\" [%s];\n"
-                   file
-                   (->> node-properties
-                        (mapcar (lambda (n)
-                                  (concat (car n) "=" "\"" (cdr n) "\"")))
-                        (s-join ","))))))
+           (format "  \"%s\" [%s];\n" file
+                   (mapconcat (lambda (n)
+                                (org-roam-graph--dot-option n nil "\""))
```

Comment:
I wonder whether `...-dot-option` should rather always wrap string-type arguments, keep numbers alone and convert symbols into unwrapped strings? This would be a breaking change towards existing configurations, though.

Reply:
I think I'll leave this for the moment, because I'm not all that familiar with the DOT language yet. Definitely could be more sophisticated if needed later on.